riscv/src/small_field/code_gen.rs

```diff
@@ -0,0 +1,1822 @@
+use std::vec;
+
+use itertools::Itertools;
+use powdr_isa_utils::SingleDataValue;
+use powdr_isa_utils::{escape_label, quote};
+
+use powdr_number::KnownField;
+
+use crate::continuations::bootloader::{bootloader_and_shutdown_routine, bootloader_preamble};
+
+use crate::code_gen::{
+    InstructionArgs, MemEntry, Register, RiscVProgram, SourceFileInfo, Statement,
+};
+use crate::CompilerOptions;
+
+use crate::small_field::runtime::Runtime;
+
+/// Translates a RISC-V program to POWDR ASM
+/// with constraints that work for a field >16 bits.
+///
+/// Will call each of the methods in the `RiscVProgram` just once.
+pub fn translate_program(program: impl RiscVProgram, options: CompilerOptions) -> String {
+    let runtime = Runtime::new(options.libs, options.continuations);
+
+    // Do this in a separate function to avoid most of the code being generic on F.
+    // TODO I think this isn't needed anymore
```

Comment:
does this todo still make sense?

Reply:
I wrote this TODO actually, because now the functions aren't generic on F anymore so we could do what the comment says. But I think it's actually fine as is so I'd just remove both comments.